File: typescript/Bidder.ts
import { Observer } from "./Observer";

// Konkrétní implementace zájemce (Sledující)
export class Bidder implements Observer {
    private name: string;

    constructor(name: string) {
        this.name = name;
    }

    public update(itemName: string, newPrice: number): void {
        console.log(` -> Sledující ${this.name} byl upozorněn: Nová cena za "${itemName}" je ${newPrice} Kč.`);
    }
}

export class Logger implements Observer {

    constructor() {
    }

    public update(itemName: string, newPrice: number): void {
        console.log(` -> Předmět "${itemName}" je ${newPrice} Kč.`);
    }
}

File: csharp/Bidder.cs
using System;

// Konkrétní implementace zájemce (Sledující)
public class Bidder : IObserver
{
    private readonly string _name;

    public Bidder(string name)
    {
        _name = name;
    }

    public void Update(string itemName, decimal newPrice)
    {
        Console.WriteLine($" -> Sledující {_name} byl upozorněn: Nová cena za \"{itemName}\" je {newPrice} Kč.");
    }
}
